napari/layers/text/text.py

```diff
@@ -0,0 +1,820 @@
+from typing import Union, List, Tuple
+from xml.etree.ElementTree import Element
+import numpy as np
+import itertools
+from copy import copy, deepcopy
+from contextlib import contextmanager
+from ..base import Layer
+from ...util.event import Event
+from ...util.misc import ensure_iterable
+from ...util.status_messages import format_float
+from vispy.color import get_color_names, Color
+from ._constants import Mode
+
+
+class Text(Layer):
+    """Annotations layer.
+
+    Parameters
+    ----------
+    data : tuple (coords, text)
+        coords contains text coordinates for N points in D dimensions.
+        text contains a list of the strings to be displayed as text.
+    rotation : float
+        Angle of the text elements
+    text_color : str
+        The color of the text font.
+    font_size : float
+        Size of the text font in points.
+    font : str
+        Text font. OpenSans is the default.
+    anchor_x : str
+        Positioning of the text relative to the coordinate. Default is 'center'.
+    anchor_y : str
+        Positioning of the text relative to the coordinate. Default is 'center'.
+    render_method : str
```

Comment:
I don't think we should expose this to our users - let's make a choice internally and hard code it as a private attribute. I'm worried about too many parameters that aren't critical for users to work with

Reply:
Good point. Centering the anchors seems to be the way to go, as it makes calculating hitboxes, etc. easy.

Regarding the render mode, here's a snipped from the vispy docs:

> The ‘cpu’ method
        should perform better on remote backends like those based on WebGL.
        The ‘gpu’ method should produce higher quality results.

I made `cpu` default for max compatibility, but it seems like for our current use cases, `gpu` is safe.

What do you all think?

Reply:
Let's go with GPU unless someone complains. We don't have plans to support WebGL for a long time. Maybe you can add that docs snippet where you set the parameter, so that we can find it easily in the future.

Reply:
Sounds good. Done!